osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs

```diff
@@ -327,5 +340,18 @@ private Vector2 getEndCursorPosition(OsuHitObject hitObject)
 
             return pos;
         }
+
+        private int getObjectCombo(HitObject hitObject)
+        {
+            if (hitObject is Slider slider)
+            {
+                if (slider.NestedHitObjects[1] is SliderRepeat)
+                    return slider.RepeatCount + 2;
+                else
+                    return slider.NestedHitObjects.Count;
+            }
```

Comment:
This code comes from [this conversation](https://discord.com/channels/546120878908506119/922996338453532722/1176545068086198373) in the pp dev discord (with an explanation of what this does). I didn't look to see if there was a better way to do this.

Reply:
I read that conversation and I don't see any actual explanation for why the nested object count isn't used? I don't see it spelled out why using `NestedHitObjects.Count` would be incorrect in that case.

The 100% correct way to do this would be to create judgements for the hitobject and all nested objects recursively, and then do `judgement.MaxResult.AffectsCombo()`. But that's probably gonna be slow, so something like this may be warranted.

Reply:
You should be doing similar to what `IBeatmap.GetMaxCombo()` does:

https://github.com/ppy/osu/blob/54288c350aee9d92f94c0dc93821487511200c8d/osu.Game/Beatmaps/IBeatmap.cs#L96-L111

Don't worry about optimisation, copy the method verbatim and adjust as necessary.

Reply:
this is what I was looking for - thanks

Reply:
Resolved in 005a5de4e196258b68c1c847000890f660ae4b68